RELEASING.rst

```diff
@@ -0,0 +1,58 @@
+Creating a release
+==================
+
+This document is documentation intended for Buildbot maintainers.
+It documents the release process of Buildbot.
+
+Step 1: Release notes PR
+------------------------
+
+Open a new branch (e.g. `release`) and run the following:
+
+    . .venv/bin/activate && make release_notes VERSION=x.y.z
+
+This collects the release notes using the `towncrier` tool and then commits the result.
+This step is done as a PR so that CI can check for spelling errors and similar issues.
+Local checks are insufficient as spelling check in particular depends on what dictionaries are installed.
+
+It's best to run `make docs` afterwards and check `master/docs/_build/html/relnotes/index.html` file for obvious rendering errors.
+This will have much faster turnaround compared to if the error is noticed after the CI runs.
+If any errors are found, just amend the commit created by `make release_notes`.
+
+Step 2: Merge the release notes PR
+----------------------------------
+
+Step 3: Perform actual release
+------------------------------
+
+This step requires the Buildbot git repository to contain `buildbot` remote that points to https://github.com/buildbot/buildbot and can be pushed to.
+
+Additionally, the Buildbot docs repository (https://github.com/buildbot/bbdocs) must be checked out at `../bbdocs` path.
+
+Pull the merge commit created on the `master` branch during the step 2.
+Then run:
+
+    make release VERSION=x.y.z
+
+This will create the required tags, make documentation, copy it to bbdocs repo and push everything.
+
+Step 4: Wait for CircleCi to create release tarballs
+----------------------------------------------------
+
+The push of tags created during step 3 will activate CircleCi configuration that generates tarballs and uploads them to GitHub.
+This is a good time to draft a release on the GitHub UI at https://github.com/buildbot/buildbot/releases.
+
+Step 5: Upload release to pypi
+------------------------------
+
+This step requires GitHub Hub tool to be installed and authorized to GitHub (https://github.com/github/hub).
+Additionally you have to have access to GPG key that is used to sign the releases.
+Finally, you have to be added as a maintainer to all Buildbot PyPi projects.
+
+To complete the release just run the following:
+
+    make finishrelease
+
+The above will download the releases from GitHub and upload them using twine.
+If you get bytes-related error after entering Pypi password, you'll need to upgrade Twine.
```

Comment:
I think you forgot the announcement part.

Write a blog post with the list of highlights of the release. A short community update is a plus to help monotony of release notes.
git shortlog also is important to recognise our contributors.

Then annoucement on the three mailing list, with link to the blog post.

Reply:
Fixed. I guess it's too late to do this for 1.8.2 and 2.3.1, isn't it?